### ISSUE TITLE
CollectionView is Very Laggy

### ISSUE THREAD

**Issue by ShariatPanah** [dotnet/maui#20433]
### Description

I'm using a vertical linear collectionview, the problem is scrolling through CollectionView is very laggy, and also navigating from a page to another is extremely slow, for example when i tap on an item in the collectionview it takes 4-5 seconds to show the requested page, Please just see the video:

https://github.com/dotnet/maui/assets/32509267/e4336a60-841f-4925-800d-3040c04a31c5

the interesting part is that this is really a very basic design and it's not complicated at all, why is this so laggy?

this video is recorded on Huawei Mate 10 Lite With Android 8.0
I tested on an Emulator With Api Level 31 (Android 12.0), the scrolling through CollectionView is smooth, but opening a page is taking something between 2-3 seconds
also tested on windows, scrolling is not laggy at all

so what is the problem?

### Version with bug

8.0.6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

**Comment by Totalq09** [comment]
Hard to tell just by looking at the video. Provide reproduction repository or at least post the XAML of that particular view.

**Comment by drasticactions** [comment]
Could you please create a small reproduction project of how you're using this view? Just looking at a video doesn't give much for what types of controls you're using and how you're using them.

**Comment by ShariatPanah** [comment]
@jsuarezruiz @samhouts @Eilon @drasticactions 
can i send the whole project to one of you in private to test it on your side?

**Comment by ShariatPanah** [comment]
here is the xaml code im using:
```
<RefreshView
    x:Name="refreshView"
    Grid.Row="3"
    Margin="{Binding IsFilteringOn, Converter={StaticResource IsFilteringOnForCollectionViewMargin}}"
    Command="{Binding RefreshCommand}"
    IsRefreshing="{Binding IsRefreshing}"
    RefreshColor="#8ABA56">
    <CollectionView
        x:Name="projectsCollectionView"
        ItemTemplate="{StaticResource projectsDataTemplate}"
        ItemsSource="{Binding Projects}"
        RemainingItemsThreshold="1"
        RemainingItemsThresholdReachedCommand="{Binding LoadMoreProjectsCommand}"
        SelectionMode="None">
        <CollectionView.ItemsLayout>
            <LinearItemsLayout ItemSpacing="15" Orientation="Vertical" />
        </CollectionView.ItemsLayout>

	<CollectionView.ItemTemplate>
<DataTemplate x:Key="projectsDataTemplate">
    <Border
        BackgroundColor="#8ABA56"
        HeightRequest="250"
        StrokeThickness="0">
        <Border.StrokeShape>
            <RoundRectangle CornerRadius="{StaticResource CornerRadius}" />
        </Border.StrokeShape>
        <Border.GestureRecognizers>
            <TapGestureRecognizer Command="{Binding Path=BindingContext.ShowProjectDetailsCommand, Source={Reference projectsCollectionView}}" CommandParameter="{Binding Id}" />
        </Border.GestureRecognizers>
        <Grid>
            <Image Aspect="AspectFill" Source="{Binding Image}" />
            <BoxView
                BackgroundColor="Transparent"
                CornerRadius="{StaticResource CornerRadius}"
                HeightRequest="60"
                Opacity="0.85"
                VerticalOptions="End"
                Color="White" />

            <Border
                HeightRequest="60"
                StrokeThickness="0"
                VerticalOptions="End">
                <Grid
                    Padding="15,5"
                    ColumnDefinitions="30,*"
                    RowDefinitions="25,20"
                    RowSpacing="5">
                    <Image
                        Grid.RowSpan="2"
                        Grid.Column="0"
                        HeightRequest="30"
                        IsVisible="{Binding IsDone}"
                        VerticalOptions="Center"
                        WidthRequest="30">
                        <Image.Source>
                            <icons:IconSource Icon="BadgeCheck" Color="White" />
                        </Image.Source>
                    </Image>

                    <Label
                        Grid.Row="0"
                        Grid.Column="1"
                        FontFamily="IranSansBold"
                        FontSize="15"
                        HorizontalOptions="End"
                        Text="{Binding Name}"
                        TextColor="White" />

                    <Label
                        Grid.Row="1"
                        Grid.Column="1"
                        FontFamily="IranSans"
                        FontSize="10"
                        HorizontalOptions="End"
                        Text="{Binding Owner}"
                        TextColor="White" />
                </Grid>
            </Border>
        </Grid>
    </Border>
</DataTemplate>
	</CollectionView.ItemTemplate>
    </CollectionView>
</RefreshView>

```


the interesting part is when i comment this line:
```<Image Aspect="AspectFill" Source="{Binding Image}" />```

the lag would gone and i can scroll through the collection view not perfectly but really smooth. so it has something to do with images in ItemTemplate

**Comment by mjsb212** [comment]
@ShariatPanah - see [18877](https://github.com/dotnet/maui/issues/18877) & https://github.com/dotnet/maui/issues/18881 - There are several open issues with CollectionView on Android loading images with certain layouts...try removing the grid and border and test in Release mode and see if the layout is causing the issue.  I'm having similar problems and when I remove the images and/or the Grid everything is fine as long as my images are not to large.

**Comment by QianaJiao** [comment]
Can repro this issue at Android platform at the latest 17.10 Preview 5 (8.0.7&8.0.21).

**Comment by jsuarezruiz** [comment]
> Can repro this issue at Android platform at the latest 17.10 Preview 5 (8.0.7&8.0.21).

Could you attach the reproduction sample?

**Comment by ShariatPanah** [comment]
i provided the xaml code i'm using, i dont think it would be too hard to copy and paste the code to your project

**Comment by matt-goldman** [comment]
I encountered this issue today after extensive debugging and found the root cause to be specifying `ItemsLayout`, specifically `LinearItemsLayout` (potentially others too but this seems to be the common thread). Here’s the snipped from your code causing the issue:

```xml
<CollectionView.ItemsLayout>
            <LinearItemsLayout ItemSpacing="15" Orientation="Vertical" />
</CollectionView.ItemsLayout>
```

From my investigation, I believe the following issues are related:

* #18877 
* #18367 
* #18395 
* #17127 
* #19383 
* #8422 

Can you confirm whether removing this resolves your issue? (Even if you need it for your layout, confirming this can help pinpoint the problem).

cc: @jsuarezruiz @drasticactions @PureWeen @jonathanpeppers - sorry for the tag spam, I can see that @ZeBrody identified the cause in #17127 (see: https://github.com/dotnet/maui/issues/17127#issuecomment-1740565116) but not sure whether this connection has been made.

**Comment by ShariatPanah** [comment]
@matt-goldman thanks for your response, i tested without setting CollectionView.ItemsLayout and honestly i don't see any difference in terms of lagging, but about ItemSpacing i can confirm that without setting it the [Continuous-related](https://github.com/dotnet/maui/issues/17127) bug would gone.

**Comment by matt-goldman** [comment]
> @matt-goldman thanks for your response, i tested without setting CollectionView.ItemsLayout and honestly i don't see any difference in terms of lagging, but about ItemSpacing i can confirm that without setting it the [Continuous-related](https://github.com/dotnet/maui/issues/17127) bug would gone.

Yep definitely appears to be related to the spacing, not the layout

**Comment by bricefriha** [comment]
I just had the same issue, I can confirm it's the spacing that causes it

**Comment by krapso** [comment]
Yes, same for me. It also causes problems for GridItemsLayout.

**Comment by Kanat9494** [comment]
> > I've been struggling with this problem for a long time too, but, more recently, I noticed that even the native apps of such giants as Telegram, Instagram, Facebook lag when scrolling large lists on my phone (Realme, OS - Android 14, RAM - 8gb), not to mention cross-platform applications. So, since then, the performance of CollectionView MAUI has become acceptable to me. If you have an average phone with OS Android, check Telegram, Instagram and Facebook app, you'll see lags when you scroll down big lists.
> 
> I see. You are right about performance issues in general, but this very issue can be avoided easily. I can tell you that for [GamHub](https://github.com/Gamhub-io/MobileApp), our main feed loads chunks of items (because there are too many items to load at once), and we still had the issue ("had" because we used the workaround above to solve this).

Wow, your app is awesome!!

I like the fact that despite some problems, people use net maui for production applications.

**Comment by ghost** [comment]
> I like the fact that despite some problems, people use net maui for production applications.

Yeah and we regret it every day ;D

**Comment by bricefriha** [comment]
> Yeah and we regret it every day ;D

@NoiseHustler I personally don't

**Comment by Shalini-Ashokan** [comment]
This issue has been verified in Visual Studio Code 1.103.2 with MAUI version 9.0.100. It was reproduced on the Android platform in both debug and release mode.

**Release**

https://github.com/user-attachments/assets/d8a1846f-0f90-45de-bcb0-d2f8eccc9731

**Debug**

https://github.com/user-attachments/assets/73ad7cf9-f06d-49a1-a75d-eeaf760b33e3
